src/transport/SecurePairingSession.cpp

```diff
@@ -0,0 +1,278 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file implements the CHIP SPAKE2P Session object that provides
+ *      APIs for constructing spake2p messages and establishing encryption
+ *      keys.
+ *
+ */
+
+#include <support/CodeUtils.h>
+#include <transport/SecurePairingSession.h>
+
+namespace chip {
+
+static const char * kSpake2pKeyExchangeSalt = "SPAKE2P Key Exchange Salt";
+
+using namespace Crypto;
+
+const char * kSpake2pI2RSessionInfo = "Commissioning I2R Key";
+const char * kSpake2pR2ISessionInfo = "Commissioning R2I Key";
+
+SecurePairingSession::SecurePairingSession(void) {}
+
+SecurePairingSession::~SecurePairingSession(void)
+{
+    if (mDelegate != nullptr)
+    {
+        mDelegate->Release();
+    }
+}
+
+CHIP_ERROR SecurePairingSession::Init(uint32_t setupCode, SecurePairingSessionDelegate * delegate)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    VerifyOrExit(delegate != NULL, err = CHIP_ERROR_INVALID_ARGUMENT);
+
+    err = mSpake2p.Init((const unsigned char *) &mSpake2pContext, sizeof(mSpake2pContext));
+    SuccessOrExit(err);
+
+    err = pbkdf2_sha256((const unsigned char *) &setupCode, sizeof(setupCode), (const unsigned char *) kSpake2pKeyExchangeSalt,
```

Comment:
We'll want the salt and iteration count to be parameters to the `SecurePairingSession ::Init` function - these parameters will be communicated to the commissioner as part of the pairing process.

Reply:
@balducci-apple this is done.